hoodie-client/src/main/java/com/uber/hoodie/io/ConsistencyCheck.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2018 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.uber.hoodie.io;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.uber.hoodie.common.SerializableConfiguration;
+import com.uber.hoodie.common.util.FSUtils;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.apache.spark.api.java.JavaSparkContext;
+
+/**
+ * Checks if all the written paths have their metadata consistent on storage and thus be listable to
+ * queries. This is important for cloud, stores like AWS S3 which are eventually consistent with
+ * their metadata. Without such checks, we may proceed to commit the written data, without the
+ * written data being made available to queries. In cases like incremental pull this can lead to
+ * downstream readers failing to ever see some data.
+ */
+public class ConsistencyCheck implements Serializable {
+
+  private static final transient Logger log = LogManager.getLogger(ConsistencyCheck.class);
+
+  private String basePath;
+
+  private List<String> relPaths;
```

Comment:
For a bootstrap tables type of commit, relPaths can be a really large number ?

Reply:
yes. someone can always turn this off though right.. if you do want consistent bootstrap, yes there is a cost.